tweak version compat w/ PHP 5.4

damonsharp opened the pull request:
For reference, see #319 & #321...

In order to send the Referer header with the Google Client API library the Google API Client library needs to be bumped from v1.8.x to v2+ (which requires PHP v5.4+) as v1.8 doesn't have the method `setHttpClient()` needed to set the Referer as follows (from `/includes/feeds/google.php`)...

```
/**
     * Google API Client.
     *
     * @since  3.0.0
     * @access private
     *
     * @return \Google_Client
     */
    private function get_client() {

        $client = new \Google_Client();
        $client->setApplicationName( 'Simple Calendar' );
        $client->setScopes( $this->google_client_scopes );
        $client->setDeveloperKey( $this->google_api_key );
        $client->setAccessType( 'online' );
        $client->setHttpClient(
            new \GuzzleHttp\Client(
                array(
                    'defaults' => array(
                        'headers' => array(
                            'Referer' => site_url(),
                        ),
                    ),
                )
            )
        );

        return $client;
    }
```
For my original solution that Cesar previously tested, the dependencies loaded via composer (Guzzle, etc.), required PHP v7.1+, but I was able to put restrictions in composer to keep these down to the 5.4+ requirement of the Google Client library as follows...

```
"require": {
    "php": ">=5.4.0",
    "ext-curl": "*",
    "ext-iconv": "*",
    "ext-json": "*",
    "ext-mbstring": "*",
    "composer/installers": "1.x",
    "google/apiclient": "~2.0",
    "nesbot/carbon": "1.x",
    "erusev/parsedown": "1.x",
    "mexitek/phpcolors": "dev-master",
    "guzzlehttp/guzzle": "*"
  },
  "config": {
    "platform": {
      "php": "5.4.0"
    }
  }
```

zachwills commented on includes/feeds/google.php:
```diff
@@ -123,6 +123,36 @@ public function __construct( $calendar = '', $load_admin = true ) {
+				array(
+					'defaults' => array(
+						'headers' => array(
+							'Referer' => site_url(),
```
Do you think we'd benefit at all by making this filterable? I know `site_url()` is filterable.. but maybe there is a use case where someone would want to only filter this specific use of it?

damonsharp commented:
@zachwills can you take one last look at this and give it an approval. added the filter and updated the version numbers and changelog. probably going to release this tomorrow evening.

zachwills commented:
@damonsharp change LGTM!
